Modules/overlapped.c

```diff
@@ -706,7 +719,9 @@ Overlapped_dealloc(OverlappedObject *self)
     }
 
     Overlapped_clear(self);
+    PyTypeObject *tp = Py_TYPE(self);
     PyObject_Del(self);
+    Py_DECREF(tp);
     SetLastError(olderr);
```

Comment:
I suggest to call SetLastError() earlier, to clarify that calling PyObject_Del(self) and Py_DECREF(tp) doesn't affect the current Windows error.

```suggestion
    Overlapped_clear(self);
    SetLastError(olderr);
    
    PyTypeObject *tp = Py_TYPE(self);
    PyObject_Del(self);
    Py_DECREF(tp);
```